### PR TITLE
Add pagination to Costmanagement Query API

### PR DESCRIPTION
**Pull request by GuillaumeColom**
# Description

Add Pagination capability to Costmanagement - Query API

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
Introduce some breaking changes by adding skipToken parameter to methods in Query API

- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/costmanagement/azure-resourcemanager-costmanagement/src/main/java/com/azure/resourcemanager/costmanagement/implementation/QueriesClientImpl.java [hunk 1]
```diff
@@ -64,6 +64,7 @@ Mono<Response<QueryResultInner>> usage(
             @HostParam("$host") String endpoint,
             @PathParam(value = "scope", encoded = true) String scope,
             @QueryParam("api-version") String apiVersion,
+            @QueryParam("$skiptoken") String skiptoken,
```
Comment (weidongxu-microsoft) [inline sdk/costmanagement/azure-resourcemanager-costmanagement/src/main/java/com/azure/resourcemanager/costmanagement/implementation/QueriesClientImpl.java:67]: I don't think costmanagement have the `$skiptoken` in their API spec https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2021-10-01/costmanagement.json#L1022-L1038 ?

This should be done in their API spec.

**Comment by GuillaumeColom** [comment]
Hello,

We are waiting another pull request to be completed (for API spec) before pursuing here.

Thank you.

**Comment by neiser** [comment]
@GuillaumeColom Thank you very much for this contribution! I highly appreciate it. The blocking PR https://github.com/Azure/azure-rest-api-specs/pull/19609 appears to have been merged now. I hope this one can then be merged as well soon. Eagerly waiting for it!

**Comment by neiser** [comment]
@weidongxu-microsoft Any chance of getting this merged soon? Do you have a timeline? Sorry for bugging you with this, but I'm actually blocked by this issue and I'm considering copying this patch manually into my project unless this gets merged/released within the next few days. So, if you already know it will take some weeks until it's merged, please let me know. Thank you!

**Comment by neiser** [comment]
@GuillaumeColom After looking into this code change in more detail, I don't think exposing `skipToken` in `com.azure.resourcemanager.costmanagement.models.Queries#usage` is not a good idea, as it's introducing a breaking change. Pagination should be handled similar how the `com.azure.resourcemanager.costmanagement.implementation.OperationsClientImpl#listAsync()` is doing it by chaining the requests.

**Comment by weidongxu-microsoft** [comment]
@neiser 

Romain is checking with cost management service to sign off on swagger https://github.com/Azure/azure-rest-api-specs/pull/19609

After it signed-off, Java will regenerate the whole SDK with 2022-08-01-preview api-version.

---

For the pageable, it is also due to swagger. If service supports it correctly, it should be annotated with `x-ms-pageable`, e.g. https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-08-01-preview/costmanagement.json#L70-L72

**Comment by Totox64** [comment]
@weidongxu-microsoft I would like to know if it's possible to reopen this PR. Since the PR in the rest api spec is merged, this sould be quite straight forward

**Comment by weidongxu-microsoft** [comment]
@Totox64 
backend decides not to change (issue here https://github.com/Azure/azure-rest-api-specs/issues/23405)

java provided a workaround here https://github.com/Azure/azure-sdk-for-java/issues/34455

**Pull request closed without merging**